cognite/client/data_classes/functions.py

```diff
@@ -134,15 +134,7 @@ def list_schedules(self, limit: int | None = DEFAULT_LIMIT_READ) -> FunctionSche
         Returns:
             FunctionSchedulesList: List of function schedules
         """
-        schedules_by_external_id = self._cognite_client.functions.schedules.list(
-            function_external_id=self.external_id, limit=limit
-        )
-        schedules_by_id = self._cognite_client.functions.schedules.list(function_id=self.id, limit=limit)
-
-        if is_unlimited(limit):
-            limit = self._cognite_client.functions.schedules._LIST_LIMIT_CEILING
-
-        return (schedules_by_external_id + schedules_by_id)[:limit]
+        return self._cognite_client.functions.schedules.list(function_id=self.id, limit=limit)
```

Comment:
According to the function docs `id` is required, so I don't see why we'd need to try getting the schedules by both external id and id, so simplified it. Although the old logic seemed to work just fine.

Reply:
I swear I got duplicates returned the other day... but I'm (also) unable to reproduce it now... I should have taken a screenshot or something... 

Either way, this is a great improvement! 🥇

Reply:
Puh 😅 😅 😅  managed to recreate the bug now:

<img width="990" alt="image" src="https://github.com/cognitedata/cognite-sdk-python/assets/8521241/70888c66-7938-4acb-9537-35527fd89e70">

Reply:
Checking out this branch, I got what I expected:

<img width="904" alt="image" src="https://github.com/cognitedata/cognite-sdk-python/assets/8521241/b1c68dfb-75b1-4e2e-993a-f073069651ea">

Reply:
Oh, I'm stupid... 🤦  When a function is missing `external_id`, you basically fetch all schedules in the project (on line `137-139`):
```py
schedules_by_external_id = self._cognite_client.functions.schedules.list(
    function_external_id=self.external_id, limit=limit
)

Reply:
fyi @HMEiding @qTipTip